woocommerce/woocommerce-ios
# Dashboard: New Orders

The PR adds the New Orders notice to the dashboard:

![new_orders](https://user-images.githubusercontent.com/154014/44614223-51b60480-a7e6-11e8-9794-f2f383d61e33.gif)![new_orders2](https://user-images.githubusercontent.com/154014/44614501-8ed0c580-a7eb-11e8-8b08-e13f6b48c017.gif)


There are a couple of moving parts to this PR:

1. I cleaned up `MainTabBarController` a little bit (making a tab `enum`) and added some static helper functions to navigate between tabs
2. `PaddedLabel`'s `@IBDesignable` designation was causing IB to crash/error for me on multiple machines when editing the Dashboard storyboard...I ended up removing them altogether
3. Pull-to-refresh works and will update the New Order notice if needed (as well as the store stats)
4. I wired up `ResultsController<StorageOrder>` to `NewOrdersViewController` to watch for changes to Orders with a "Processed" status — I am only using the count here, so we will probably want to optimize ResultsController/Storage at a later date

Fixes: #35 

## Testing

As a baseline, make sure the app builds, runs and all the unit tests are ✅ 

Next, try a few different scenarios:

* Open a site with 0 orders in processing status, verify the notice _doesn't_ display; pull-to-refresh verify the notice _doesn't_ display
* Open a site with 1 order in processing status, verify the notice _does_ display (and is singular)
* Open a site with 2+ orders in processing status, verify the notice _does_ display (and is plural)
* Open a site with 0 orders in processing status, verify the notice _doesn't_ display, update an order with processing status on the web site, pull-to-refresh in-app, and verify the notice _does_ display

...make sure tapping the notice works as expected too. In short, try and break this! 

Also note that since we are using `ResultsController` on this screen, any updates made on the Orders screens OR via "Undo" in the note cards _should_ be immediately reflected here.   

@mindgraffiti @jleandroperez Would you mind checking this one out for me? Thanks! 😄
